dali/operators/generic/reduce/reduce.cc

```diff
@@ -29,6 +29,10 @@ DALI_SCHEMA(ReduceBase)
 
 Not providing any axis results in reduction of all elements.)code",
     std::vector<int>{ })
+  .AddOptionalArg("axis_names", R"code(Axis or axes along which reduction is performed, given by name.
+
+Axis indices are taken from the input layout, and this argument cannot be used with ``axes``.)code",
+    TensorLayout(""))
```

Comment:
I think that we should have a clear distinction between "empty" (which means "not reducing at all" or just be an error) and "not provided". I'd recommend ```c++
AddOptionalArg<TensorLayout>("axis_names", "...", nullptr)```

Reply:
Done